src/Microsoft.ComponentDetection.Detectors/go/Go117ComponentDetector.cs

```diff
@@ -0,0 +1,204 @@
+
+            try
+            {
+                if (goModFile == null)
+                {
+                    this.Logger.LogDebug(
+                        "go.sum file found without an adjacent go.mod file. Location: {Location}",
+                        processRequest.ComponentStream.Location);
+
+                    return true;
+                }
+
+                return GoDetectorUtils.ShouldRemoveGoSumFromDetection(goSumFilePath: processRequest.ComponentStream.Location, goModFile, this.Logger);
+            }
+            finally
+            {
+                goModFile?.Stream.Dispose();
+            }
+        });
+
+        return Task.FromResult(goModProcessRequests);
+    }
+
+    protected override async Task OnFileFoundAsync(ProcessRequest processRequest, IDictionary<string, string> detectorArgs, CancellationToken cancellationToken = default)
+    {
+        var singleFileComponentRecorder = processRequest.SingleFileComponentRecorder;
+        var file = processRequest.ComponentStream;
+
+        var projectRootDirectory = Directory.GetParent(file.Location);
+        if (this.projectRoots.Any(path => projectRootDirectory.FullName.StartsWith(path)))
+        {
+            return;
+        }
+
+        var record = new GoGraphTelemetryRecord();
+        var fileExtension = Path.GetExtension(file.Location).ToUpperInvariant();
+        switch (fileExtension)
+        {
+            case ".MOD":
+            {
+                this.Logger.LogDebug("Found Go.mod: {Location}", file.Location);
+                await this.goParserFactory.CreateParser(GoParserType.GoMod, this.Logger).ParseAsync(singleFileComponentRecorder, file, record);
+
+                if (await this.ShouldRunGoGraphAsync())
+                {
+                    await GoDependencyGraphUtility.GenerateAndPopulateDependencyGraphAsync(
+                        this.commandLineInvocationService,
+                        this.Logger,
+                        singleFileComponentRecorder,
+                        projectRootDirectory.FullName,
+                        record,
+                        cancellationToken);
+                }
+
+                break;
+            }
+
+            case ".SUM":
+            {
+                this.Logger.LogDebug("Found Go.sum: {Location}", file.Location);
+                await this.goParserFactory.CreateParser(GoParserType.GoSum, this.Logger).ParseAsync(singleFileComponentRecorder, file, record);
+                break;
+            }
+
+            default:
+            {
+                throw new InvalidOperationException("Unexpected file type detected in go detector");
+            }
+        }
+    }
+
+    private async Task<bool> ShouldRunGoGraphAsync()
+    {
+        var goVersion = await this.GetGoVersionAsync();
+        if (goVersion == null)
+        {
+            return false;
+        }
+
+        return goVersion >= new Version(1, 11);
```

Comment:
We should also respect the [DisableGoCliScan](https://github.com/microsoft/component-detection/blob/main/docs/detectors/go.md#fallback-detection-strategy) variable here, if customer explicitly asked for no CLI, then we should not generate the graph (edges) between the nodes regardless of Go CLI Version. They will see flat list of components and may be fine for performance reasons they just won't be able to determine root module that brought in the dependency.